## Content list

#### 1. Import libraries
#### 2. Import data
#### 3. Check data set
#### 4. Checking mixed type columns
#### 5. Addressing missing value
#### 6. Checking for duplicates
#### 7. New columns creations / modifications
#### 8. Checking data statistics
#### 9. Export data

## 1. Import libraries

In [107]:
# Import libraries
import pandas as pd
import numpy as np
import os
import datetime as dt

## 2. Import Data

In [2]:
# Turn project folder path into a string
path = r'C:\Users\Admin\Desktop\Data Analysis\Advanced Analytics & Dashboard Design\Competitive_LoL_Analysis'

In [3]:
path

'C:\\Users\\Admin\\Desktop\\Data Analysis\\Advanced Analytics & Dashboard Design\\Competitive_LoL_Analysis'

In [5]:
# Import league competitive data
df_lol = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'League_of_Legends_Competitive_Data_2024.csv'), index_col = False)

In [6]:
# Check output
df_lol.head()

gameid league  year             date  game  patch  participantid  \
0  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              1   
1  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              2   
2  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              3   
3  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              4   
4  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              5   

   side position  champion  ... kills deaths assists teamkills teamdeaths  \
0  Blue      top  Renekton  ...     3      3       3        10         14   
1  Blue      jng  Nocturne  ...     2      2       2        10         14   
2  Blue      mid     Akali  ...     2      3       1        10         14   
3  Blue      bot   Kalista  ...     2      2       2        10         14   
4  Blue      sup      Pyke  ...     1      4       3        10         14   

  damagetochampions damageshare totalgold golddiffat15 xpdiffat15  
0             10063    0.286116     12202       1690.0     -346.0  
1              4562    0.129709      9892       -378.0       74.0  
2             11408    0.324358     10279        279.0     -330.0  
3              6014    0.170993     11703        213.0     1319.0  
4              3124    0.088823      7580        -75.0    -1313.0  

[5 rows x 32 columns]

## 3. Check Data Set

In [8]:
# Print the name of columns in df_lol
df_lol.columns

Index(['gameid', 'league', 'year', 'date', 'game', 'patch', 'participantid',
       'side', 'position', 'champion', 'ban1', 'ban2', 'ban3', 'ban4', 'ban5',
       'pick1', 'pick2', 'pick3', 'pick4', 'pick5', 'gamelength', 'result',
       'kills', 'deaths', 'assists', 'teamkills', 'teamdeaths',
       'damagetochampions', 'damageshare', 'totalgold', 'golddiffat15',
       'xpdiffat15'],
      dtype='object')

In [9]:
# Print number of rows and columns in df_lol
df_lol.shape

(10512, 32)

In [10]:
# Print data type for each columns in df_lol
df_lol.dtypes

gameid                object
league                object
year                   int64
date                  object
game                   int64
patch                float64
participantid          int64
side                  object
position              object
champion              object
ban1                  object
ban2                  object
ban3                  object
ban4                  object
ban5                  object
pick1                 object
pick2                 object
pick3                 object
pick4                 object
pick5                 object
gamelength             int64
result                 int64
kills                  int64
deaths                 int64
assists                int64
teamkills              int64
teamdeaths             int64
damagetochampions      int64
damageshare          float64
totalgold              int64
golddiffat15         float64
xpdiffat15           float64
dtype: object

## 4. Checking for Mixed Type Columns

In [11]:
# Check for mixed types in columns
for col in df_lol.columns.tolist():
  weird = (df_lol[[col]].map(type) != df_lol[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_lol[weird]) > 0:
    print (col)

champion
ban3
ban4
ban5
pick1
pick2
pick3
pick4
pick5


In [17]:
# Changing mix column to string
df_lol['champion'] = df_lol['champion'].astype('str')
df_lol['ban3'] = df_lol['ban3'].astype('str')
df_lol['ban4'] = df_lol['ban4'].astype('str')
df_lol['ban5'] = df_lol['ban5'].astype('str')
df_lol['pick1'] = df_lol['pick1'].astype('str')
df_lol['pick2'] = df_lol['pick2'].astype('str')
df_lol['pick3'] = df_lol['pick3'].astype('str')
df_lol['pick4'] = df_lol['pick4'].astype('str')
df_lol['pick5'] = df_lol['pick5'].astype('str')

In [18]:
# Check again for mixed types in columns
for col in df_lol.columns.tolist():
  weird = (df_lol[[col]].map(type) != df_lol[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_lol[weird]) > 0:
    print (col)

## 5. Addressing Missing Value

In [19]:
# Check for missing value in df_ords
df_lol.isnull().sum()

gameid                  0
league                  0
year                    0
date                    0
game                    0
patch                  24
participantid           0
side                    0
position                0
champion                0
ban1                    0
ban2                    0
ban3                    0
ban4                    0
ban5                    0
pick1                   0
pick2                   0
pick3                   0
pick4                   0
pick5                   0
gamelength              0
result                  0
kills                   0
deaths                  0
assists                 0
teamkills               0
teamdeaths              0
damagetochampions       0
damageshare          1752
totalgold               0
golddiffat15         4500
xpdiffat15           4500
dtype: int64

## 5.1 Addressing missing value for 'patch' column

In [24]:
# Create subset of dataframe that countains the missing value in 'patch'
df_lol_patch = df_lol[df_lol['patch'].isnull() == True]

In [25]:
df_lol_patch.head(24)

gameid league  year             date  game  patch  \
8904  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8905  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8906  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8907  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8908  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8909  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8910  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8911  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8912  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8913  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8914  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8915  10932-10932_game_1    LPL  2024  27/3/2024 11:17     1    NaN   
8916  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8917  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8918  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8919  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8920  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8921  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8922  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8923  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8924  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8925  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8926  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   
8927  10932-10932_game_2    LPL  2024  27/3/2024 12:07     2    NaN   

      participantid  side position  champion  ... kills deaths assists  \
8904              1  Blue      top    Aatrox  ...     5      2       5   
8905              2  Blue      jng   Lee Sin  ...     1      3       3   
8906              3  Blue      mid     Karma  ...     2      0       9   
8907              4  Blue      bot     Senna  ...     4      2       7   
8908              5  Blue      sup  Nautilus  ...     2      2       9   
8909              6   Red      top  Renekton  ...     2      5       3   
8910              7   Red      jng        Vi  ...     4      2       4   
8911              8   Red      mid   Taliyah  ...     2      3       5   
8912              9   Red      bot   Smolder  ...     1      1       3   
8913             10   Red      sup     Braum  ...     0      3       7   
8914            100  Blue     team       nan  ...    14      9      33   
8915            200   Red     team       nan  ...     9     14      22   
8916              1  Blue      top   Rek'Sai  ...     0      4       8   
8917              2  Blue      jng       Jax  ...     6      2       4   
8918              3  Blue      mid      Hwei  ...     3      2       8   
8919              4  Blue      bot   Smolder  ...     4      1       7   
8920              5  Blue      sup  Nautilus  ...     0      3      10   
8921              6   Red      top  Renekton  ...     6      2       5   
8922              7   Red      jng  Xin Zhao  ...     0      3       7   
8923              8   Red      mid     Karma  ...     3      2       9   
8924              9   Red      bot     Senna  ...     2      5       9   
8925             10   Red      sup      Ornn  ...     1      1       8   
8926            100  Blue     team       nan  ...    13     12      37   
8927            200   Red     team       nan  ...    12     13      38   

     teamkills teamdeaths damagetochampions damageshare totalgold  \
8904        14          9             23159    0.270146     13053   
8905        14          9              4347    0.050713      9459   
8906        14          9             29275    0.341488     11069   
8907        14          9             20639    0.240758      9441   
8908        14          9              8306   

#### Seems like all the missing patch version is from date 27/3/2024, i can cross check the games with similar dates and add impute in the patch version

In [42]:
# Create subset of dataframe that countains the dates the different patch version and check the date range
df_lol_patch_ver1 = df_lol[df_lol['patch'] == 14.01]
df_lol_patch_ver2 = df_lol[df_lol['patch'] == 14.02]
df_lol_patch_ver3 = df_lol[df_lol['patch'] == 14.03]
df_lol_patch_ver4 = df_lol[df_lol['patch'] == 14.04]
df_lol_patch_ver5 = df_lol[df_lol['patch'] == 14.05]
df_lol_patch_ver6 = df_lol[df_lol['patch'] == 14.06]

In [46]:
with pd.option_context('display.max_rows', None):
  display(df_lol_patch_ver1['date'].value_counts)

<bound method IndexOpsMixin.value_counts of 0       13/1/2024 16:10
1       13/1/2024 16:10
2       13/1/2024 16:10
3       13/1/2024 16:10
4       13/1/2024 16:10
5       13/1/2024 16:10
6       13/1/2024 16:10
7       13/1/2024 16:10
8       13/1/2024 16:10
9       13/1/2024 16:10
10      13/1/2024 16:10
11      13/1/2024 16:10
12      13/1/2024 17:06
13      13/1/2024 17:06
14      13/1/2024 17:06
15      13/1/2024 17:06
16      13/1/2024 17:06
17      13/1/2024 17:06
18      13/1/2024 17:06
19      13/1/2024 17:06
20      13/1/2024 17:06
21      13/1/2024 17:06
22      13/1/2024 17:06
23      13/1/2024 17:06
24      13/1/2024 18:15
25      13/1/2024 18:15
26      13/1/2024 18:15
27      13/1/2024 18:15
28      13/1/2024 18:15
29      13/1/2024 18:15
30      13/1/2024 18:15
31      13/1/2024 18:15
32      13/1/2024 18:15
33      13/1/2024 18:15
34      13/1/2024 18:15
35      13/1/2024 18:15
36      13/1/2024 19:24
37      13/1/2024 19:24
38      13/1/2024 19:24
39      13/1/2024 19

In [47]:
with pd.option_context('display.max_rows', None):
  display(df_lol_patch_ver2['date'].value_counts)

<bound method IndexOpsMixin.value_counts of 1404    27/1/2024 20:15
1405    27/1/2024 20:15
1406    27/1/2024 20:15
1407    27/1/2024 20:15
1408    27/1/2024 20:15
1409    27/1/2024 20:15
1410    27/1/2024 20:15
1411    27/1/2024 20:15
1412    27/1/2024 20:15
1413    27/1/2024 20:15
1414    27/1/2024 20:15
1415    27/1/2024 20:15
1416    27/1/2024 21:14
1417    27/1/2024 21:14
1418    27/1/2024 21:14
1419    27/1/2024 21:14
1420    27/1/2024 21:14
1421    27/1/2024 21:14
1422    27/1/2024 21:14
1423    27/1/2024 21:14
1424    27/1/2024 21:14
1425    27/1/2024 21:14
1426    27/1/2024 21:14
1427    27/1/2024 21:14
1428    27/1/2024 22:29
1429    27/1/2024 22:29
1430    27/1/2024 22:29
1431    27/1/2024 22:29
1432    27/1/2024 22:29
1433    27/1/2024 22:29
1434    27/1/2024 22:29
1435    27/1/2024 22:29
1436    27/1/2024 22:29
1437    27/1/2024 22:29
1438    27/1/2024 22:29
1439    27/1/2024 22:29
1440    27/1/2024 23:22
1441    27/1/2024 23:22
1442    27/1/2024 23:22
1443    27/1/2024 23

In [48]:
with pd.option_context('display.max_rows', None):
  display(df_lol_patch_ver3['date'].value_counts)

<bound method IndexOpsMixin.value_counts of 3072    10/2/2024 21:07
3073    10/2/2024 21:07
3074    10/2/2024 21:07
3075    10/2/2024 21:07
3076    10/2/2024 21:07
3077    10/2/2024 21:07
3078    10/2/2024 21:07
3079    10/2/2024 21:07
3080    10/2/2024 21:07
3081    10/2/2024 21:07
3082    10/2/2024 21:07
3083    10/2/2024 21:07
3084    10/2/2024 22:01
3085    10/2/2024 22:01
3086    10/2/2024 22:01
3087    10/2/2024 22:01
3088    10/2/2024 22:01
3089    10/2/2024 22:01
3090    10/2/2024 22:01
3091    10/2/2024 22:01
3092    10/2/2024 22:01
3093    10/2/2024 22:01
3094    10/2/2024 22:01
3095    10/2/2024 22:01
3096    10/2/2024 22:53
3097    10/2/2024 22:53
3098    10/2/2024 22:53
3099    10/2/2024 22:53
3100    10/2/2024 22:53
3101    10/2/2024 22:53
3102    10/2/2024 22:53
3103    10/2/2024 22:53
3104    10/2/2024 22:53
3105    10/2/2024 22:53
3106    10/2/2024 22:53
3107    10/2/2024 22:53
3108    10/2/2024 23:44
3109    10/2/2024 23:44
3110    10/2/2024 23:44
3111    10/2/2024 23

In [49]:
with pd.option_context('display.max_rows', None):
  display(df_lol_patch_ver4['date'].value_counts)

<bound method IndexOpsMixin.value_counts of 5028      1/3/2024 7:17
5029      1/3/2024 7:17
5030      1/3/2024 7:17
5031      1/3/2024 7:17
5032      1/3/2024 7:17
5033      1/3/2024 7:17
5034      1/3/2024 7:17
5035      1/3/2024 7:17
5036      1/3/2024 7:17
5037      1/3/2024 7:17
5038      1/3/2024 7:17
5039      1/3/2024 7:17
5040      1/3/2024 8:09
5041      1/3/2024 8:09
5042      1/3/2024 8:09
5043      1/3/2024 8:09
5044      1/3/2024 8:09
5045      1/3/2024 8:09
5046      1/3/2024 8:09
5047      1/3/2024 8:09
5048      1/3/2024 8:09
5049      1/3/2024 8:09
5050      1/3/2024 8:09
5051      1/3/2024 8:09
5076      1/3/2024 9:46
5077      1/3/2024 9:46
5078      1/3/2024 9:46
5079      1/3/2024 9:46
5080      1/3/2024 9:46
5081      1/3/2024 9:46
5082      1/3/2024 9:46
5083      1/3/2024 9:46
5084      1/3/2024 9:46
5085      1/3/2024 9:46
5086      1/3/2024 9:46
5087      1/3/2024 9:46
5100     1/3/2024 10:37
5101     1/3/2024 10:37
5102     1/3/2024 10:37
5103     1/3/2024 10

In [50]:
with pd.option_context('display.max_rows', None):
  display(df_lol_patch_ver5['date'].value_counts)

<bound method IndexOpsMixin.value_counts of 5976     8/3/2024 21:07
5977     8/3/2024 21:07
5978     8/3/2024 21:07
5979     8/3/2024 21:07
5980     8/3/2024 21:07
5981     8/3/2024 21:07
5982     8/3/2024 21:07
5983     8/3/2024 21:07
5984     8/3/2024 21:07
5985     8/3/2024 21:07
5986     8/3/2024 21:07
5987     8/3/2024 21:07
5988     8/3/2024 21:52
5989     8/3/2024 21:52
5990     8/3/2024 21:52
5991     8/3/2024 21:52
5992     8/3/2024 21:52
5993     8/3/2024 21:52
5994     8/3/2024 21:52
5995     8/3/2024 21:52
5996     8/3/2024 21:52
5997     8/3/2024 21:52
5998     8/3/2024 21:52
5999     8/3/2024 21:52
6000     8/3/2024 22:42
6001     8/3/2024 22:42
6002     8/3/2024 22:42
6003     8/3/2024 22:42
6004     8/3/2024 22:42
6005     8/3/2024 22:42
6006     8/3/2024 22:42
6007     8/3/2024 22:42
6008     8/3/2024 22:42
6009     8/3/2024 22:42
6010     8/3/2024 22:42
6011     8/3/2024 22:42
6012     8/3/2024 23:32
6013     8/3/2024 23:32
6014     8/3/2024 23:32
6015     8/3/2024 23

#### Seems like the patch version that is for date 27/3/2024 is 14.05

In [51]:
# Set those rows with empty patch version to 14.05
df_lol.loc[df_lol['patch'].isnull() == True, 'patch'] = 14.05

In [52]:
# Check if patch column still have null value
df_lol['patch'].isnull().sum()

0

## 5.2 Addressing missing value for 'damageshare' column

In [53]:
# Create subset of dataframe that countains the missing value in 'damageshare'
df_lol_dmgshare = df_lol[df_lol['damageshare'].isnull() == True]

In [54]:
df_lol_dmgshare.head()

gameid league  year             date  game  patch  participantid  \
10  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01            100   
11  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01            200   
22  LOLTMNT06_12701    LEC  2024  13/1/2024 17:06     1  14.01            100   
23  LOLTMNT06_12701    LEC  2024  13/1/2024 17:06     1  14.01            200   
34  LOLTMNT06_13667    LEC  2024  13/1/2024 18:15     1  14.01            100   

    side position champion  ... kills deaths assists teamkills teamdeaths  \
10  Blue     team      nan  ...    10     14      11        10         14   
11   Red     team      nan  ...    14     10      26        14         10   
22  Blue     team      nan  ...     4     16      13         4         16   
23   Red     team      nan  ...    16      4      43        16          4   
34  Blue     team      nan  ...     9     20      16         9         20   

   damagetochampions damageshare totalgold golddiffat15 xpdiffat15  
10             35171         NaN     51656       1729.0     -596.0  
11             66376         NaN     56807      -1729.0      596.0  
22             58731         NaN     55881       -331.0      392.0  
23             68283         NaN     68768        331.0     -392.0  
34            107413         NaN     66367       -461.0      298.0  

[5 rows x 32 columns]

In [56]:
df_lol_dmgshare.tail()

gameid league  year             date  game  patch  \
10487  10956-10956_game_2    LPL  2024  20/4/2024 10:11     2  14.06   
10498  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10499  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10510  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10511  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   

       participantid  side position champion  ... kills deaths assists  \
10487            200   Red     team      nan  ...    22      4      63   
10498            100  Blue     team      nan  ...    22     21      52   
10499            200   Red     team      nan  ...    21     22      46   
10510            100  Blue     team      nan  ...    16      4      41   
10511            200   Red     team      nan  ...     4     16      12   

      teamkills teamdeaths damagetochampions damageshare totalgold  \
10487        22          4             68774         NaN     51113   
10498        22         21            100827         NaN     72782   
10499        21         22            131640         NaN     66911   
10510        16          4            108319         NaN     63209   
10511         4         16             68686         NaN     49032   

      golddiffat15 xpdiffat15  
10487          NaN        NaN  
10498          NaN        NaN  
10499          NaN        NaN  
10510          NaN        NaN  
10511          NaN        NaN  

[5 rows x 32 columns]

In [57]:
# Checking if damage share value is null due to the column 'position' representing team
df_lol_dmgshare['position'].value_counts()

position
team    1752
Name: count, dtype: int64

#### Seems like all the null values are contains for team, as this columns is for the individual damage breakdown percentage, it is not applicable for team, thus it is fine to leave it as null.

## 5.3 Addressing missing value for 'golddiffat15' column

In [58]:
# Create subset of dataframe that countains the missing value in 'golddiffat15'
df_lol_golddiff = df_lol[df_lol['golddiffat15'].isnull() == True]

In [59]:
df_lol_golddiff.head(20)

gameid league  year             date  game  patch  \
660  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
661  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
662  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
663  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
664  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
665  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
666  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
667  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
668  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
669  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
670  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
671  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
672  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
673  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
674  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
675  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
676  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
677  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
678  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
679  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   

     participantid  side position      champion  ... kills deaths assists  \
660              1  Blue      top       K'Sante  ...     1      0       6   
661              2  Blue      jng         Brand  ...     8      0       8   
662              3  Blue      mid       LeBlanc  ...     4      0       6   
663              4  Blue      bot         Varus  ...     1      2       8   
664              5  Blue      sup          Ashe  ...     3      1      10   
665              6   Red      top          Udyr  ...     1      2       1   
666              7   Red      jng      Xin Zhao  ...     0      5       1   
667              8   Red      mid          Azir  ...     1      3       0   
668              9   Red      bot       Kalista  ...     0      3       3   
669             10   Red      sup      Nautilus  ...     1      4       2   
670            100  Blue     team           nan  ...    17      3      38   
671            200   Red     team           nan  ...     3     17       7   
672              1  Blue      top          Udyr  ...     2      2       6   
673              2  Blue      jng     Jarvan IV  ...     2      0       8   
674              3  Blue      mid          Azir  ...     1      0       6   
675              4  Blue      bot       Kalista  ...     8      2       5   
676              5  Blue      sup  Renata Glasc  ...     2      2      10   
677              6   Red      top        Aatrox  ...     0      2       2   
678              7   Red      jng            Vi  ...     1      3       2   
679              8   Red      mid       Orianna  ...     3      3       2   

    teamkills teamdeaths damagetochampions damageshare totalgold golddiffat15  \
660        17          3             10197    0.129578     10913          NaN   
661        17          3             18832    0.239303     12311          NaN   
662        17          3             26196    0.332882     13156          NaN   
663        17          3             12877    0.163639     11495          NaN   
664        17          3             10592    0.134599      9104          NaN   
665         3         17              8595    0.208143      9972          NaN   
666         3         17              7823    0.189433      7385          NaN   
667         3         17             14914    0.361155      8131          NaN   
668         3         17              6621    0.160336      9343          NaN   
669         3         17              3342    0.080933      5718          NaN   
670        17          3             

In [60]:
df_lol_golddiff.tail(20)

gameid league  year             date  game  patch  \
10492  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10493  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10494  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10495  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10496  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10497  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10498  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10499  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10500  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10501  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10502  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10503  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10504  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10505  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10506  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10507  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10508  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10509  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10510  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10511  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   

       participantid  side position      champion  ... kills deaths assists  \
10492              5  Blue      sup          Ashe  ...     2      4      15   
10493              6   Red      top        Rumble  ...     3      4       6   
10494              7   Red      jng            Vi  ...     2      4      10   
10495              8   Red      mid  Aurelion Sol  ...     2      3       9   
10496              9   Red      bot         Varus  ...    11      2       8   
10497             10   Red      sup       Kalista  ...     3      9      13   
10498            100  Blue     team           nan  ...    22     21      52   
10499            200   Red     team           nan  ...    21     22      46   
10500              1  Blue      top      Renekton  ...     2      1       9   
10501              2  Blue      jng       Kindred  ...     5      0       4   
10502              3  Blue      mid          Azir  ...     2      1       9   
10503              4  Blue      bot         Varus  ...     5      1       9   
10504              5  Blue      sup         Neeko  ...     2      1      10   
10505              6   Red      top        Gragas  ...     0      3       2   
10506              7   Red      jng            Vi  ...     0      5       3   
10507              8   Red      mid       Taliyah  ...     1      2       2   
10508              9   Red      bot       Kalista  ...     3      4       1   
10509             10   Red      sup         Rakan  ...     0      2       4   
10510            100  Blue     team           nan  ...    16      4      41   
10511            200   Red     team           nan  ...     4     16      12   

      teamkills teamdeaths damagetochampions damageshare totalgold  \
10492        22         21             18610    0.184572     10539   
10493        21         22             30403    0.230956     12525   
10494        21         22             13951    0.105981     11509   
10495        21         22             25795    0.195949     14979   
10496        21         22             39826    0.302537     16762   
10497        21         22             21665    0.164577     11136   
10498        22         21            100827         NaN     72782   
10499        21         22            131640         NaN     66911   
10500        16          4             19945    0.184136     11808   
10501        16          4             18587    0.171596     12953   
10502        16          4             27496    0.253840     14509   
10503 

In [61]:
# Checking if golddiffat15 value is null due to league in 'LPL'
df_lol_golddiff['league'].value_counts()

league
LPL    4500
Name: count, dtype: int64

In [62]:
# Checking if all LPL does not have golddiff data.
df_lol_lpl = df_lol[df_lol['league'] == 'LPL']

In [66]:
# Checking shape of matches with LPL data
df_lol_lpl.shape

(4500, 32)

In [67]:
df_lol_lpl['golddiffat15'].count()

0

#### Seems like all LPL matches do not have data on golddiffat15. I would continue to leave this blank and would exclude LPL data from analysis if any analysis required that column.

## 5.4 Addressing missing value for 'xpdiffat15' column

In [68]:
# Create subset of dataframe that countains the missing value in 'xpdiffat15'
df_lol_xpdiff = df_lol[df_lol['xpdiffat15'].isnull() == True]

In [69]:
df_lol_xpdiff.head(20)

gameid league  year             date  game  patch  \
660  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
661  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
662  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
663  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
664  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
665  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
666  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
667  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
668  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
669  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
670  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
671  10665-10665_game_1    LPL  2024   22/1/2024 9:24     1  14.01   
672  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
673  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
674  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
675  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
676  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
677  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
678  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   
679  10665-10665_game_2    LPL  2024  22/1/2024 10:14     2  14.01   

     participantid  side position      champion  ... kills deaths assists  \
660              1  Blue      top       K'Sante  ...     1      0       6   
661              2  Blue      jng         Brand  ...     8      0       8   
662              3  Blue      mid       LeBlanc  ...     4      0       6   
663              4  Blue      bot         Varus  ...     1      2       8   
664              5  Blue      sup          Ashe  ...     3      1      10   
665              6   Red      top          Udyr  ...     1      2       1   
666              7   Red      jng      Xin Zhao  ...     0      5       1   
667              8   Red      mid          Azir  ...     1      3       0   
668              9   Red      bot       Kalista  ...     0      3       3   
669             10   Red      sup      Nautilus  ...     1      4       2   
670            100  Blue     team           nan  ...    17      3      38   
671            200   Red     team           nan  ...     3     17       7   
672              1  Blue      top          Udyr  ...     2      2       6   
673              2  Blue      jng     Jarvan IV  ...     2      0       8   
674              3  Blue      mid          Azir  ...     1      0       6   
675              4  Blue      bot       Kalista  ...     8      2       5   
676              5  Blue      sup  Renata Glasc  ...     2      2      10   
677              6   Red      top        Aatrox  ...     0      2       2   
678              7   Red      jng            Vi  ...     1      3       2   
679              8   Red      mid       Orianna  ...     3      3       2   

    teamkills teamdeaths damagetochampions damageshare totalgold golddiffat15  \
660        17          3             10197    0.129578     10913          NaN   
661        17          3             18832    0.239303     12311          NaN   
662        17          3             26196    0.332882     13156          NaN   
663        17          3             12877    0.163639     11495          NaN   
664        17          3             10592    0.134599      9104          NaN   
665         3         17              8595    0.208143      9972          NaN   
666         3         17              7823    0.189433      7385          NaN   
667         3         17             14914    0.361155      8131          NaN   
668         3         17              6621    0.160336      9343          NaN   
669         3         17              3342    0.080933      5718          NaN   
670        17          3             

In [70]:
df_lol_xpdiff.tail(20)

gameid league  year             date  game  patch  \
10492  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10493  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10494  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10495  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10496  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10497  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10498  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10499  10956-10956_game_3    LPL  2024  20/4/2024 11:02     3  14.06   
10500  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10501  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10502  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10503  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10504  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10505  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10506  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10507  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10508  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10509  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10510  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   
10511  10956-10956_game_4    LPL  2024  20/4/2024 12:03     4  14.06   

       participantid  side position      champion  ... kills deaths assists  \
10492              5  Blue      sup          Ashe  ...     2      4      15   
10493              6   Red      top        Rumble  ...     3      4       6   
10494              7   Red      jng            Vi  ...     2      4      10   
10495              8   Red      mid  Aurelion Sol  ...     2      3       9   
10496              9   Red      bot         Varus  ...    11      2       8   
10497             10   Red      sup       Kalista  ...     3      9      13   
10498            100  Blue     team           nan  ...    22     21      52   
10499            200   Red     team           nan  ...    21     22      46   
10500              1  Blue      top      Renekton  ...     2      1       9   
10501              2  Blue      jng       Kindred  ...     5      0       4   
10502              3  Blue      mid          Azir  ...     2      1       9   
10503              4  Blue      bot         Varus  ...     5      1       9   
10504              5  Blue      sup         Neeko  ...     2      1      10   
10505              6   Red      top        Gragas  ...     0      3       2   
10506              7   Red      jng            Vi  ...     0      5       3   
10507              8   Red      mid       Taliyah  ...     1      2       2   
10508              9   Red      bot       Kalista  ...     3      4       1   
10509             10   Red      sup         Rakan  ...     0      2       4   
10510            100  Blue     team           nan  ...    16      4      41   
10511            200   Red     team           nan  ...     4     16      12   

      teamkills teamdeaths damagetochampions damageshare totalgold  \
10492        22         21             18610    0.184572     10539   
10493        21         22             30403    0.230956     12525   
10494        21         22             13951    0.105981     11509   
10495        21         22             25795    0.195949     14979   
10496        21         22             39826    0.302537     16762   
10497        21         22             21665    0.164577     11136   
10498        22         21            100827         NaN     72782   
10499        21         22            131640         NaN     66911   
10500        16          4             19945    0.184136     11808   
10501        16          4             18587    0.171596     12953   
10502        16          4             27496    0.253840     14509   
10503 

In [71]:
# Checking if golddiffat15 value is null due to league in 'LPL'
df_lol_xpdiff['league'].value_counts()

league
LPL    4500
Name: count, dtype: int64

In [72]:
# Checking for xpdiffat15 count form the LPL subset data
df_lol_lpl['xpdiffat15'].count()

0

#### Seems like all LPL matches do not have data on xpdiffat15. I would continue to leave this blank and would exclude LPL data from analysis if any analysis required that column.

## 5.5 Cross checking on data set count to filter any missing value

In [76]:
## Checking dataset summary
df_lol.count()

gameid               10512
league               10512
year                 10512
date                 10512
game                 10512
patch                10512
participantid        10512
side                 10512
position             10512
champion             10512
ban1                 10512
ban2                 10512
ban3                 10512
ban4                 10512
ban5                 10512
pick1                10512
pick2                10512
pick3                10512
pick4                10512
pick5                10512
gamelength           10512
result               10512
kills                10512
deaths               10512
assists              10512
teamkills            10512
teamdeaths           10512
damagetochampions    10512
damageshare           8760
totalgold            10512
golddiffat15          6012
xpdiffat15            6012
dtype: int64

#### Seems like those columns with numerical values all have 10,512 rows, with the exception of damageshare, golddiffat15 and xpdiffat15.

## 6. Checking for duplicates

In [77]:
# Check for duplicate in df_lol
df_lol_dups = df_lol[df_lol.duplicated()]

In [78]:
df_lol_dups

Empty DataFrame
Columns: [gameid, league, year, date, game, patch, participantid, side, position, champion, ban1, ban2, ban3, ban4, ban5, pick1, pick2, pick3, pick4, pick5, gamelength, result, kills, deaths, assists, teamkills, teamdeaths, damagetochampions, damageshare, totalgold, golddiffat15, xpdiffat15]
Index: []

[0 rows x 32 columns]

#### No duplicates found.

## 7. New column creation / modifications

## 7.1 Creating 'kda' column [(kill +_assist) / death]

In [91]:
# Creating new kda columns where kda = (kills + assists) / deaths
df_lol['kda'] = (df_lol.kills + df_lol.assists) / df_lol.deaths

In [84]:
# Check if kda columns calculates correctly
with pd.option_context('display.max_columns', None):
  display(df_lol.head(5))

gameid league  year             date  game  patch  participantid  \
0  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              1   
1  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              2   
2  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              3   
3  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              4   
4  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              5   

   side position  champion    ban1      ban2     ban3 ban4   ban5 pick1 pick2  \
0  Blue      top  Renekton  Draven  Bel'Veth  LeBlanc  Jax  Poppy   nan   nan   
1  Blue      jng  Nocturne  Draven  Bel'Veth  LeBlanc  Jax  Poppy   nan   nan   
2  Blue      mid     Akali  Draven  Bel'Veth  LeBlanc  Jax  Poppy   nan   nan   
3  Blue      bot   Kalista  Draven  Bel'Veth  LeBlanc  Jax  Poppy   nan   nan   
4  Blue      sup      Pyke  Draven  Bel'Veth  LeBlanc  Jax  Poppy   nan   nan   

  pick3 pick4 pick5  gamelength  result  kills  deaths  assists  teamkills  \
0   nan   nan   nan        1749       0      3       3        3         10   
1   nan   nan   nan        1749       0      2       2        2         10   
2   nan   nan   nan        1749       0      2       3        1         10   
3   nan   nan   nan        1749       0      2       2        2         10   
4   nan   nan   nan        1749       0      1       4        3         10   

   teamdeaths  damagetochampions  damageshare  totalgold  golddiffat15  \
0          14              10063     0.286116      12202        1690.0   
1          14               4562     0.129709       9892        -378.0   
2          14              11408     0.324358      10279         279.0   
3          14               6014     0.170993      11703         213.0   
4          14               3124     0.088823       7580         -75.0   

   xpdiffat15  kda  
0      -346.0  2.0  
1        74.0  2.0  
2      -330.0  1.0  
3      1319.0  2.0  
4     -1313.0  1.0

In [97]:
# Check if all rows have kda count
df_lol['kda'].count()

10512

In [93]:
# Create subset of dataframe that countains the missing value in 'kda'
df_lol_kda = df_lol[df_lol['kda'].isnull() == True]

In [95]:
# Check if kda columns calculates correctly
with pd.option_context('display.max_columns', None):
  display(df_lol_kda.head())

gameid league  year            date  game  patch  \
1464  LOLTMNT02_34441    LCK  2024  28/1/2024 7:00     2  14.01   

      participantid  side position champion     ban1     ban2   ban3 ban4  \
1464              1  Blue      top     Udyr  LeBlanc  K'Sante  Rakan  Jax   

      ban5 pick1 pick2 pick3 pick4 pick5  gamelength  result  kills  deaths  \
1464  Ashe   nan   nan   nan   nan   nan        2373       0      0       0   

      assists  teamkills  teamdeaths  damagetochampions  damageshare  \
1464        0          3           3              12445     0.190865   

      totalgold  golddiffat15  xpdiffat15  kda region  
1464      12049           1.0        20.0  NaN  Korea

#### Seems like it is showing blank as the calcuation is showing as 0/0, thus i will have to manually impute the value of 0 in.

In [96]:
# Set the row with empty kda to 0
df_lol.loc[df_lol['kda'].isnull() == True, 'kda'] = 0

In [99]:
# Check if all rows have kda count again
df_lol['kda'].count()

10512

In [120]:
df_lol['kda'].value_counts()

kda
inf          1089
1.000000      571
2.000000      429
3.000000      352
4.000000      281
             ... 
2.111111        1
0.300000        1
33.000000       1
3.320000        1
3.045455        1
Name: count, Length: 857, dtype: int64

#### Seems like there is some columns that have infinite value, might be due to dividing by 0. To fix this issue, i will let the columns calculation for KDA to divide by 1 if death is 0.

In [123]:
# Set the kda rows with 0 deaths to divide by 1 instead of 0
df_lol.loc[df_lol['deaths'] == 0, 'kda'] = (df_lol.kills + df_lol.assists)

In [124]:
# Checking if inf values still exist
df_lol['kda'].value_counts()

kda
1.000000     578
2.000000     444
3.000000     366
4.000000     303
5.000000     294
            ... 
17.333333      1
0.227273       1
38.000000      1
1.421053       1
3.045455       1
Name: count, Length: 861, dtype: int64

## 7.2 Creating 'region' column

In [86]:
# Creating region columns and setting regions based on the competitve league
# LCK = Korea, LCS = North America, LPL = China, LEC = Europe
df_lol.loc[df_lol['league'] == 'LCK', 'region'] = 'Korea'
df_lol.loc[df_lol['league'] == 'LCS', 'region'] = 'North America'
df_lol.loc[df_lol['league'] == 'LPL', 'region'] = 'China'
df_lol.loc[df_lol['league'] == 'LEC', 'region'] = 'Europe'

In [87]:
# Check if region column is created properly
df_lol.head()

gameid league  year             date  game  patch  participantid  \
0  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              1   
1  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              2   
2  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              3   
3  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              4   
4  LOLTMNT06_13630    LEC  2024  13/1/2024 16:10     1  14.01              5   

   side position  champion  ... assists teamkills teamdeaths  \
0  Blue      top  Renekton  ...       3        10         14   
1  Blue      jng  Nocturne  ...       2        10         14   
2  Blue      mid     Akali  ...       1        10         14   
3  Blue      bot   Kalista  ...       2        10         14   
4  Blue      sup      Pyke  ...       3        10         14   

  damagetochampions damageshare totalgold golddiffat15 xpdiffat15  kda  region  
0             10063    0.286116     12202       1690.0     -346.0  2.0  Europe  
1              4562    0.129709      9892       -378.0       74.0  2.0  Europe  
2             11408    0.324358     10279        279.0     -330.0  1.0  Europe  
3              6014    0.170993     11703        213.0     1319.0  2.0  Europe  
4              3124    0.088823      7580        -75.0    -1313.0  1.0  Europe  

[5 rows x 34 columns]

In [100]:
# Check if all rows have kda count again
df_lol['region'].count()

10512

In [103]:
# Check if all region row have a proper value
df_lol['region'].value_counts()

region
China            4500
Korea            2904
Europe           2076
North America    1032
Name: count, dtype: int64

## 7.3 Conversion of 'date' columns to show just date

In [112]:
# Convert date column to datetime format
df_lol['date'] = pd.to_datetime(df_lol['date'])

In [115]:
# Convert date column to just show date
df_lol['date'] = df_lol['date'].dt.date

In [116]:
# Checking if columns show dates correctly
df_lol.head()

gameid league  year        date  game  patch  participantid  side  \
0  LOLTMNT06_13630    LEC  2024  2024-01-13     1  14.01              1  Blue   
1  LOLTMNT06_13630    LEC  2024  2024-01-13     1  14.01              2  Blue   
2  LOLTMNT06_13630    LEC  2024  2024-01-13     1  14.01              3  Blue   
3  LOLTMNT06_13630    LEC  2024  2024-01-13     1  14.01              4  Blue   
4  LOLTMNT06_13630    LEC  2024  2024-01-13     1  14.01              5  Blue   

  position  champion  ... assists teamkills teamdeaths damagetochampions  \
0      top  Renekton  ...       3        10         14             10063   
1      jng  Nocturne  ...       2        10         14              4562   
2      mid     Akali  ...       1        10         14             11408   
3      bot   Kalista  ...       2        10         14              6014   
4      sup      Pyke  ...       3        10         14              3124   

  damageshare totalgold golddiffat15 xpdiffat15  kda  region  
0    0.286116     12202       1690.0     -346.0  2.0  Europe  
1    0.129709      9892       -378.0       74.0  2.0  Europe  
2    0.324358     10279        279.0     -330.0  1.0  Europe  
3    0.170993     11703        213.0     1319.0  2.0  Europe  
4    0.088823      7580        -75.0    -1313.0  1.0  Europe  

[5 rows x 34 columns]

## 8. Checking data statisitics

In [117]:
# Check for count for all columns.
df_lol.count()

gameid               10512
league               10512
year                 10512
date                 10512
game                 10512
patch                10512
participantid        10512
side                 10512
position             10512
champion             10512
ban1                 10512
ban2                 10512
ban3                 10512
ban4                 10512
ban5                 10512
pick1                10512
pick2                10512
pick3                10512
pick4                10512
pick5                10512
gamelength           10512
result               10512
kills                10512
deaths               10512
assists              10512
teamkills            10512
teamdeaths           10512
damagetochampions    10512
damageshare           8760
totalgold            10512
golddiffat15          6012
xpdiffat15            6012
kda                  10512
region               10512
dtype: int64

In [125]:
# Check for basic stats of data set
df_lol.describe()

year          game         patch  participantid    gamelength  \
count  10512.0  10512.000000  10512.000000   10512.000000  10512.000000   
mean    2024.0      1.757991     14.033470      29.583333   1957.325342   
std        0.0      0.909591      0.018147      57.653210    338.849700   
min     2024.0      1.000000     14.010000       1.000000   1264.000000   
25%     2024.0      1.000000     14.020000       3.750000   1713.750000   
50%     2024.0      2.000000     14.040000       6.500000   1901.500000   
75%     2024.0      2.000000     14.050000       9.250000   2166.000000   
max     2024.0      5.000000     14.060000     200.000000   3309.000000   

             result         kills        deaths       assists     teamkills  \
count  10512.000000  10512.000000  10512.000000  10512.000000  10512.000000   
mean       0.500000      4.291476      4.301560     10.524353     12.874429   
std        0.500024      5.265647      5.002806     12.608995      6.716958   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      1.000000      1.000000      3.000000      7.000000   
50%        0.500000      2.000000      3.000000      7.000000     13.000000   
75%        1.000000      5.000000      5.000000     12.000000     18.000000   
max        1.000000     38.000000     38.000000     95.000000     38.000000   

         teamdeaths  damagetochampions  damageshare      totalgold  \
count  10512.000000       10512.000000  8760.000000   10512.000000   
mean      12.904680       25406.205289     0.200000   19870.481925   
std        6.708739       26299.377131     0.097600   18630.829125   
min        0.000000        1254.000000     0.023581    4765.000000   
25%        7.000000        9063.000000     0.116409    9881.000000   
50%       13.000000       16088.000000     0.198288   12546.000000   
75%       18.000000       27356.500000     0.271810   16299.750000   
max       38.000000      203097.000000     0.600699  105904.000000   

       golddiffat15   xpdiffat15           kda  
count   6012.000000  6012.000000  10512.000000  
mean       0.000000     0.000000      5.375413  
std     1182.837726  1016.997144      5.528282  
min    -8624.000000 -5625.000000      0.000000  
25%     -550.000000  -505.000000      1.416667  
50%        0.000000     0.000000      3.500000  
75%      550.000000   505.000000      8.000000  
max     8624.000000  5625.000000     84.000000

In [127]:
# Checking stats for date column for earliest date
df_lol['date'].min()

datetime.date(2024, 1, 13)

In [128]:
# Checking stats for date column for latest date
df_lol['date'].max()

datetime.date(2024, 4, 20)

## 9. Export Data

In [132]:
# Exporting df_lol as 'competitve_data_cleaned.csv'
df_lol.to_csv(os.path.join(path, '02 Data','Prepared Data', 'competitve_data_cleaned.csv'), index=False)